sky/backends/cloud_vm_ray_backend.py

```diff
@@ -886,13 +886,16 @@ def _try_provision_tpu(self, to_provision: resources_lib.Resources,
             logger.error(stderr)
             raise e
 
-    def _retry_region_zones(self,
-                            to_provision: resources_lib.Resources,
-                            num_nodes: int,
-                            dryrun: bool,
-                            stream_logs: bool,
-                            cluster_name: str,
-                            cluster_exists: bool = False):
+    def _retry_region_zones(
+        self,
+        to_provision: resources_lib.Resources,
+        num_nodes: int,
+        dryrun: bool,
+        stream_logs: bool,
+        cluster_name: str,
+        cluster_exists: bool = False,
+        task: task_lib.Task = None,
```

Comment:
In what case, the task will not be passed or set to `None`? If there is no case, let's remove the default value for it and move it to before L893

Reply:
There a few cases such as the tests/test_spot.py